hail_search/queries/mito.py

```diff
@@ -112,6 +118,215 @@ class MitoHailTableQuery(BaseHailTableQuery):
+    ) -> tuple[hl.Table, hl.Table]:
+        sample_types = set(project_sample_type_data.keys())
+        if len(sample_types) == 1:
+            return super()._import_and_filter_entries_ht(project_guid, num_families, project_sample_type_data, **kwargs)
+
+        self._has_both_sample_types = True
+        entries_hts_map: dict[str, list[tuple[hl.Table, dict]]] = {}
+        for sample_type in sample_types:
+            if num_families == 1:
+                family_guid = list(list(project_sample_type_data.values())[0].keys())[0]
+                ht, sample_data = self._load_prefiltered_family_ht(family_guid, sample_type, project_sample_type_data, **kwargs)
+            else:
+                ht, sample_data = self._load_prefiltered_project_ht(project_guid, sample_type, project_sample_type_data, **kwargs)
+            entries_hts_map[sample_type] = [(ht, sample_data)]
+
+        return self._filter_entries_ht_both_sample_types(entries_hts_map, **kwargs)
+
+    def _filter_entries_ht_both_sample_types(
+        self, entries_hts_map: dict[str, list[tuple[hl.Table, dict]]], n_partitions=MAX_PARTITIONS,
+        inheritance_filter=None, quality_filter=None, **kwargs
+    ):
+        filtered_hts = []
+        filtered_comp_het_hts = []
+        wes_entries, wgs_entries = entries_hts_map[SAMPLE_TYPE_WES], entries_hts_map[SAMPLE_TYPE_WGS]  # entries_hts_map: {<sample_type>: [(ht, project_samples), ...]}
+
+        # Iterate over each chunk of project tables
+        for (wes_ht, wes_project_samples), (wgs_ht, wgs_project_samples) in zip(wes_entries, wgs_entries):
+            wes_ht, sorted_wes_family_sample_data = self._add_entry_sample_families(wes_ht, wes_project_samples, SAMPLE_TYPE_WES)
+            wgs_ht, sorted_wgs_family_sample_data = self._add_entry_sample_families(wgs_ht, wgs_project_samples, SAMPLE_TYPE_WGS)
+            family_idx_map = self._build_family_index_map(sorted_wes_family_sample_data, sorted_wgs_family_sample_data)
+
+            wes_ht = wes_ht.rename({'family_entries': WES_FAMILY_ENTRIES_FIELD, 'filters': WES_FILTERS_FIELD})
+            wgs_ht = wgs_ht.rename({'family_entries': WGS_FAMILY_ENTRIES_FIELD, 'filters': WGS_FILTERS_FIELD})
+            ht = wes_ht.join(wgs_ht, how='outer')
+
+            # Filter by quality
+            ht = self._annotate_passes_quality_filter(ht, quality_filter, **kwargs)
+
+            # Filter by inheritance
+            any_valid_entry, is_any_affected = self._get_any_family_member_gt_has_alt_filter()
+            ht = ht.annotate(wes_passes_inheritance=ht[WES_FAMILY_ENTRIES_FIELD].map(
+                lambda entries: hl.or_missing(entries.any(any_valid_entry), entries)
+            ))
+            ht = ht.annotate(wgs_passes_inheritance=ht[WGS_FAMILY_ENTRIES_FIELD].map(
+                lambda entries: hl.or_missing(entries.any(any_valid_entry), entries)
+            ))
+
+            if self._has_comp_het_search:
```

Comment:
the control flow of this function looks nearly identical to that of the original `_filter_inheritance`. Instead of needing to copy-past that function here and then replicate that logic, its probably better to leave the control flow implemented once in that function and change the behavior of the helpers called by that function (like `_filter_families_inheritance`). I think the rough implementation in this class would look something like this:

```
def _filter_families_inheritance(self, ht, inheritance_mode, inheritance_filter, sorted_family_sample_data):
    if len(sorted_family_sample_data) == 1:
        return super()._filter_families_inheritance(ht, inheritance_mode, inheritance_filter, sorted_family_sample_data[0])

     ht = self._annotate_families_inheritance(ht, inheritance_mode, inheritance_filter, sorted_family_sample_data[0], SAMPLE_TYPE_WES)
     ht = self._annotate_families_inheritance(ht, inheritance_mode, inheritance_filter, sorted_family_sample_data[1], SAMPLE_TYPE_WES)
     return self._apply_filters(ht, family_idx_map)
```

Reply:
Yeah the entire flow for both sample types is veeery similar to the one in base.py but it's hard to make the base.py functions generic enough to handle both sample types without explicitly referencing sample types.

The divergence from the base.py filtering occurs [here](https://github.com/broadinstitute/seqr/pull/4330/files#diff-71932dea9e59d8d1ce70d88444658acffce2f3865c9afb11b57f93daf6610f66R155-R166) in `_import_and_filter_multiple_project_hts`. After that, all of the functions are similarly named and have similar behavior but the both sample types ones are suffixed "both_sample_types".

I will take another stab at this to see if I can share code between functions that have very similar logic.